### PR TITLE
Show bottom navigation when called from custom menu entry (fix #14552)

### LINKED ISSUE #14552: Bottom navigation done wrong - problems with custom bottom navigation items
### Describe your problem!

I like the idea of having the last bottom navigation icon customizable, however it is done completely wrong. 

### How to reproduce?

- set the last item e.g. to "settings" 

### Actual result after these steps?

When clicking on the last item the bottom navigation disappears. But even worse, there is no way to get back. When using the back button cgeo gets closed completely. 

### Expected result after these steps?

- Bottom navigation should always stay visible after using it

- c:geo should guide to the dashboard first before exiting 

### Reproducible

Yes

### c:geo Version

Since July 2023 (nightly) 

### System information

_No response_

### Additional Information

Introduced in  #14419

### PR DESCRIPTION
**Pull request by moving-bits**
## Description
Keep bottom navigation when item called via custom bottom navigation menu entry, but don't show it when called via menu or quicklaunch item.

### CODE REVIEW

**Comment by fm-sys** [comment]
Wouldn't it make sense to introduce some kind of general `ICustomMenuEntry` or `CustomMenuEntryActivity` abstraction level? But maybe I'm overabstracting things now. It's difficult to look at code on a mobile, so I cannot really tell...

In any way, I would set the default to `HIDE_NAVIGATION` for these kind of activities. Then it's just necessary to trigger the "show" only at one single place.

**Comment by moving-bits** [comment]
Ok, I added a `CustomMenuEntryActivity` which defaults to "bottom navigation hidden" and assigned activities accordingly.

**Review by fm-sys — COMMENTED** [review]
Hey, thank you! May I suggest some more naming suggestions?

File: main/src/main/java/cgeo/geocaching/activity/AbstractNavigationBarActivity.java [hunk 1]
```diff
@@ -310,14 +314,29 @@ private static void launchActivity(final Activity fromActivity, final int id) {
...
-                QuickLaunchItem.launchQuickLaunchItem(fromActivity, item);
+                QuickLaunchItem.launchQuickLaunchItem(fromActivity, item, false);
                 return;
             }
         }
         final Intent launchIntent = getBottomNavigationIntent(fromActivity, id);
         fromActivity.startActivity(launchIntent);
     }
 
+    public static void setIntentHideBottomNavigation(final Intent intent, final boolean hideBottomNavigation) {
```
[the remark this answers is not in the document]
Comment (moving-bits) [reply main/src/main/java/cgeo/geocaching/activity/AbstractNavigationBarActivity.java:325]: Well, yes. I've renamed not only those two methods now, but a couple of more places where we were still using bottom navigation.

**Pull request merged**